app/services/foreman/unattended_installation/mac_list_extractor.rb

```diff
@@ -0,0 +1,31 @@
+module Foreman
+  module UnattendedInstallation
+    class MacListExtractor
+      def extract_from_env(env, params: {})
+        mac_list = macs_from_headers(env)
+        mac_list << params[:mac].strip.downcase if params[:mac]
+
+        mac_list
+      end
+
+      private
+
+      def macs_from_headers(env)
+        return [] unless env['HTTP_X_RHN_PROVISIONING_MAC_0'].present?
+
+        # Search for a mac address in any of the RHN provisioning headers.
+        begin
+          mac_list = env.keys.map do |header|
+            env[header].split[1].strip.downcase if header =~ /^HTTP_X_RHN_PROVISIONING_MAC_/
+          end
+        rescue => e
+          puts e
```

Comment:
Debug code?

Reply:
I guess this should have been a call to `logger.warn`